### ISSUE TITLE
Better Carthage support

### ISSUE THREAD

**Issue by irace** [CocoaPods/pod-template#104]
The currently pod template doesn’t create a framework target in the “main” project, only in the Pods project. This makes it a bit fiddly to support Carthage (not that CocoaPods should necessarily be bending over backwards, but an easier way to support both systems would be great).

Carthage currently looks for a framework target in the root-most Xcode project, which would not be the Pods project. You could theoretically create a shared scheme in the sample app Xcode project that builds the framework target from the Pods project, though this would result in the build artifact being called `Pod-YourLibraryName.framework` (possibly not the end of the world).

In short, I don’t have any great suggestions (I’ve just been manually creating Xcode projects with framework targets in lieu of using `pod lib create` for the time being) but wanted to at least get my thoughts down.

**Comment by orta** [comment]
Got an example?

**Comment by irace** [comment]
> Got an example?

An example of what specifically?

**Comment by orta** [comment]
Of a working library that does this

**Comment by irace** [comment]
That does this? “I’ve just been manually creating Xcode projects with framework targets in lieu of using pod lib create for the time being”

**Comment by orta** [comment]
I mean a repo I can go and look at, that has this kind of behaviour.

**Comment by irace** [comment]
I don’t have an example of this (it’s what I tried but eventually gave up):

> You could theoretically create a shared scheme in the sample app Xcode project that builds the framework target from the Pods project, though this would result in the build artifact being called Pod-YourLibraryName.framework (possibly not the end of the world).

I do have an example of this (https://github.com/tumblr/XExtensionItem): 

> I’ve just been manually creating Xcode projects with framework targets in lieu of using `pod lib create` for the time being

**Comment by alloy** [comment]
@irace It’s probably a good idea for the template to create something more universal :+1: I think what would be helpful is a repo that has a check-in of what the current template generates and the a check-in of the changes you had to manually make.

**Comment by neonichu** [comment]
@irace did you try if the shared scheme actually works? That seems like at least a tangible improvement on the current state of things.

**Comment by neonichu** [comment]
FWIW, simply sharing the Pods project schemes doesn't work for me and there seems to be no way to debug this except for building a custom version of Carthage.

**Comment by irace** [comment]
I'm not near a computer right now. You'd need to create a new scheme in the Example project, but have that scheme build the Pods project's framework target. I'm not even sure this would work though.

**Comment by orta** [comment]
#105 gets us almost all the way there now, and we know a technique to massage carthage support with new libs
